src/runtime/policy/ingress-validation.ts

```diff
@@ -162,6 +162,40 @@ export class IngressValidation {
     return this.policies.some(policy => policy.targets.some(
       target => target.schemaName === type.getEntitySchema().name));
   }
+
+  // Get the max readable type for `typeName` according to the given `policies`.
+  static getMaxReadType(typeName: string, policies: Policy[]): Type|null {
+    const fields = {};
+    let type = null;
+    for (const policy of policies) {
+      for (const target of policy.targets) {
+        if (typeName === target.schemaName) {
+          type = target.type;
+          IngressValidation.mergeFields(fields, target.getRestrictedFields());
+        }
+      }
+    }
+    return type ? EntityType.make([typeName], fields, type.getEntitySchema()) : null;
```

Comment:
I see you take: description, refinements and annotations (third argument to EntityType.make) from the last encountered type. What is the reasoning behind this vs. not using the third argument at all and let those be empty?

Reply:
This is obsolete.